### PR TITLE
Mobile app fixes

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- [x] Resolves forecast data not showing on insights.
- [x] Adds padding to insights visualizations.
- [x] Adds more tests for insights page.
- [x] Removes unused tests.
- [x] Fixes date time translations.
- [x] Resolves iOS build issues.
- [x] Disables widget updating if platform is not android.
- [x] Fixes rating bugs

#### Status of maturity (all need to be checked before merging):

- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [x] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?
To run dev
```shell
task run
```
To build for release
iOS
```shell
task build-ios
```
Android
```shell
task buid-android
```

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

### CODE REVIEW

File: mobile/lib/utils/extensions.dart [hunk 1]
```diff
@@ -580,23 +563,10 @@ extension DateTimeExt on DateTime {
...
-      throw UnimplementedError(
-        '$weekday does not have a weekday string implementation',
-      );
-    }
-
-    return weekdays[weekday - 1];
+  String getWeekday(BuildContext context) {
+    final locale = Localizations.localeOf(context);
+    final dateFormat = DateFormat.EEEE(locale.toString());
+    return dateFormat.format(this);
```
Comment (NoahNsimbe) [inline mobile/lib/utils/extensions.dart:569]: @katendejericho5 date translations.

File: mobile/test/profile_widgets_test.dart [hunk 2]
```diff
@@ -1,41 +1,81 @@
...
 void main() {
   testWidgets('Test whether sign up button has text', (tester) async {
+    final key = GlobalKey();
     await tester.pumpWidget(
-      const MaterialApp(
-        home: SignUpButton(),
+      MaterialApp(
+        localizationsDelegates: const [
+          AppLocalizations.delegate,
+          GlobalMaterialLocalizations.delegate,
```
Comment (NoahNsimbe) [inline mobile/test/profile_widgets_test.dart:14]: Adding localisation to tests. cc @katendejericho5

File: mobile/test/screens/insights/insights_widgets_test.dart [hunk 3]
```diff
@@ -77,26 +91,38 @@ void main() {
...
+            Locale('fr'), // French
+          ],
+          home: InsightAirQualityWidget(emptyInsight, name: name, key: key),
         ),
       );
 
       final nameFinder = find.text(name);
-      final airQualityTextFinder = find.text('No air quality data available');
+      final airQualityTextFinder = find.text(
+          AppLocalizations.of(key.currentContext!)!.noAirQualityDataAvailable);
```
Comment (NoahNsimbe) [inline mobile/test/screens/insights/insights_widgets_test.dart:125]: Using localisation for insight widget tests

File: mobile/Taskfile.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      - flutter clean
+    cmds:
+      - flutter build ipa --flavor=airqo --obfuscate --split-debug-info=${PWD}/obfuscate
+      - defer: { task: ios-build-cleanup }
+
+  ios-build-cleanup:
+    desc: Performs cleaning after building iOS
+    dotenv: ['.env.prod']
+    cmds:
+      - sed -i '' "s/$IOS_GOOGLE_API_KEY/GOOGLE_API_KEY/g" ios/Runner/AppDelegate.swift
```
Comment (NoahNsimbe) [inline mobile/Taskfile.yml:45]: Task to build iOS ipa

File: mobile/lib/utils/extensions.dart (hunk 1 above)
Comment (katendejericho5) [reply mobile/lib/utils/extensions.dart:569]: resolved

File: mobile/lib/utils/extensions.dart [hunk 5]
```diff
@@ -580,23 +563,10 @@ extension DateTimeExt on DateTime {
...
-      throw UnimplementedError(
-        '$weekday does not have a weekday string implementation',
-      );
-    }
-
-    return weekdays[weekday - 1];
+  String getWeekday(BuildContext context) {
+    final locale = Localizations.localeOf(context);
+    final dateFormat = DateFormat.EEEE(locale.toString());
+    return  AppLocalizations.of(context)!.getWeekday(dateFormat.format(this));
```
Comment (NoahNsimbe) [inline mobile/lib/utils/extensions.dart:569]: return dateFormat.format(this); @katendejericho5

File: mobile/lib/services/widget_service.dart [hunk 6]
```diff
@@ -83,8 +86,10 @@ class WidgetService {
   }
 
   static Future<void> sendAndUpdate() async {
-    await sendData();
-    await updateWidget();
+    if (Platform.isAndroid) {
+      await sendData();
+      await updateWidget();
+    }
```
Comment (NoahNsimbe) [inline mobile/lib/services/widget_service.dart:92]: Updating widget data only when the platform is android.

File: mobile/lib/services/location_service.dart [hunk 7]
```diff
@@ -54,6 +54,16 @@ class LocationService {
...
   static Future<CurrentLocation?> getCurrentLocation() async {
+    final locationGranted = await LocationService.locationGranted();
+    if (!locationGranted) {
+      return null;
+    }
+
+    final bool serviceEnabled = await Geolocator.isLocationServiceEnabled();
+    if (!serviceEnabled) {
+      return null;
+    }
```
Comment (NoahNsimbe) [inline mobile/lib/services/location_service.dart:65]: checking whether location services are available before attempting to get a user's location.

File: mobile/test/utils/rating_test.dart [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import 'package:app/models/models.dart';
+import 'package:app/utils/utils.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('Should not rate for new users', () {
+    bool showRating = Profile.initialize().requiresRating();
+    expect(showRating, false);
+  });
```
Comment (NoahNsimbe) [inline mobile/test/utils/rating_test.dart:9]: rating tests

**Review by BenjaminSsempala — APPROVED** [review]

**Review by katendejericho5 — APPROVED** [review]
LGTM

**Review by katendejericho5 — APPROVED** [review]
LGTM

**Pull request merged**